### DISCUSSION TITLE
Please help me understand.

### DISCUSSION THREAD

**Discussion by mikethetreeclimber** [livewire/livewire#3372] [1. Help] [answered]
I was having an error of... 
"Property [$] not found on component: [trees.tree-characteristic-form]   **stack trace**:  vendor/livewire/livewire/src/Component.php
Livewire\Component :230
 But it was only when any of the first four values were checked or unchecked whether there was 20 values dynamically added or 5 values like in the code below. Any value indexed greater then 3 does not throw the exception and is pushed to the array when checked   

CODE as follows

```
**CLASS**
class TreeCharacteristicForm extends Component
{
   
    public $treeSpecialValues;
    public $selectedSpecialValues = [];


**CHECKBOX INPUT THAT WORKS **
@foreach ($treeSpecialValues as $specialValue)

        <input 
            type="checkbox" 
            wire:model="selectedSpecialValues"
            value="{{ $specialValue->id }}"
            name="{{ $specialValue->type }}"
            id="option{{ $specialValue->id }}"
            />
       

**DOESNT WORK**
@foreach ($treeSpecialValues as $key => $specialValue)
    
        <input type="checkbox" wire:model="selectedSpecialValues"
            value="{{ $specialValue->id }}"
            name="{{ $specialValue->type }}{{ $key }}"
            id="option{{ $key }}"
        />
      
```

**Comment by Dimonka2** [comment, marked as answer]
Very long post.. 
So where I see the problem is that your checkboxes are connected to the property `wire:model="selectedSpecialValues"`, not to the own index inside `selectedSpecialValues[]`. You need to link checkboxes more like wire:model="selectedSpecialValues.{{ $specialValue->id }}" and you can ignore (do not fill out) `name` and `value` of checkbox definition. What you really required to do is to pre fill `$this->selectedSpecialValues[]` with currently selected options.
I hope my comment is making sense.

**Reply by mikethetreeclimber** [reply]
*I really appreciate the feedback.*
If I understand you right...

```<input 
      type="checkbox" 
      wire:model="selectedSpecialValues.{{ $specialValue->id }}"
```     
is the best way to wire:model values that are all being pushed to the same array?  

What had me scratching my head was the fact that only the first 4 values were throwing an error and all other values were pushed to the array and was even able to submit the form and collect every value except the first four.  I even removed the first 4 values for *$treeSpecialValues* but the the actual values had no impact just the position of the values.  I am just trying to understand what exactly was causing the error.

**Reply by Dimonka2** [reply]
> is the best way to wire:model values that are all being pushed to the same array?

If your list is made of unknown number of items - yes.  Otherwise you can bind checkbox to a particular single model property.

Regarding the "working" "non-working" code - hard to say - for me both versions do not make sense from the Livewire standpoint.
